src/code/z_actor.c

```diff
@@ -158,9 +165,43 @@ void Actor_SetScale(Actor* actor, f32 scale) {
     actor->scale.x = scale;
 }
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_SetObjectSegment.asm")
+void Actor_SetObjectSegment(GlobalContext* ctxt, Actor* actor) {
+    gRspSegmentPhysAddrs[6] = (s32) ctxt->sceneContext.objects[actor->objectIndex].vramAddr + 0x80000000;
+}
+
+#ifdef NON_MATCHING
+void Actor_InitToDefaultValues(Actor* actor, GlobalContext* ctxt) {
+    Actor_InitCurrPosition(actor);
+    Actor_InitDrawRotation(actor);
+    Actor_SetHeight(actor, 0);
+    Lib_CopyVec3f(&actor->lastPos, &actor->currPosRot.pos);
+    Actor_SetScale(actor, 0.01);
+
+    actor->pad1F[0] = 3;
+
+    actor->minYVelocity = -20.0f;
+
+    actor->naviMsgId = 255;
+
+    actor->sqrdDistToLink = D_801DCA54;
+
+    actor->unkFC = 1000.0f;
+    actor->unk100 = 350.0f;
+    actor->unk104 = 700.0f;
+    actor->meshAttachedTo = 0x32;
 
+    func_800E7494(&actor->unkA0);
+    Actor_SetDrawParams(&actor->drawParams, 0, 0, 0);
+    if (Scene_IsObjectLoaded(&ctxt->sceneContext, actor->objectIndex) != 0)
+    {
```

Comment:
We generally put braces on the same line as the condition
```
if (Scene_IsObjectLoaded(&ctxt->sceneContext, actor->objectIndex) != 0) {
// functional code
}
```

Reply:
Fixed